sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/EqualPartitionsBalancingStrategy.java

```diff
@@ -78,6 +78,13 @@ public List<Lease> selectLeasesToTake(List<Lease> allLeases) {
                 }
             }
 
+            // If we reach here with partitionsNeededForMe < 0, then it means the change feed processor instances has owned leases more than the maxScaleCount.
+            // Then in this case, the change feed processor instance will not pick up any new leases.
+            if (partitionsNeededForMe < 0)
```

Comment:
what about `partitionsNeededForMe == 0`
Though it will be automatically be covered below, but do we want to include here or not conceptually?

Reply:
I would also prefer if we change this to "<= 0" with the change in the comment as well that we reach this case where current instance has either more or exact "maxScaleCount" number of leases owned.